dc-cudami-server/dc-cudami-server-backend-jdbi/src/main/java/de/digitalcollections/cudami/server/backend/impl/database/migration/V9_02_02__DML_Fill_urlaliases.java

```diff
@@ -148,33 +148,76 @@ private void migrateWebpages(JdbcTemplate jdbcTemplate) throws SQLException {
           JSONObject jsonObject = new JSONObject(w.toString());
           UUID uuid = UUID.fromString(jsonObject.getString("w_uuid"));
           UUID websiteUuid = UUID.fromString(jsonObject.getString("ws_uuid"));
-          try {
-            Map<String, String> labels =
-                new ObjectMapper().readValue(jsonObject.getString("label"), HashMap.class);
-            labels.forEach(
-                (language, label) -> {
-                  UrlAlias urlAlias =
-                      buildUrlAlias(
-                          jdbcTemplate,
-                          null,
-                          IdentifiableType.RESOURCE,
-                          uuid,
-                          language,
-                          label,
-                          websiteUuid);
-                  try {
-                    saveUrlAlias(jdbcTemplate, urlAlias);
-                  } catch (SQLException e) {
-                    throw new RuntimeException("Cannot save urlAlias " + urlAlias + ": " + e, e);
-                  }
-                });
-          } catch (JsonProcessingException e) {
-            throw new RuntimeException("Cannot parse " + w + ": " + e, e);
-          }
+          createUrlAliasAndMigrateSubpages(
+              jdbcTemplate, webpages, w, jsonObject, uuid, websiteUuid);
         });
     LOGGER.info("Successfully added {} UrlAliases for webpages", webpages.size());
   }
 
+  private void migrateSubpages(JdbcTemplate jdbcTemplate, UUID websiteUuid, UUID parentWebpageUuid)
+      throws SQLException {
+    String selectQuery =
+        "SELECT w.uuid AS w_uuid, w.label AS label FROM webpages w, webpage_webpages ww WHERE ww.parent_webpage_uuid = '"
+            + parentWebpageUuid
+            + "' and ww.child_webpage_uuid=w.uuid";
```

Comment:
Dieser Select ist sehr implizit (geht natürlich) und ich hab wirklich eine Weile gebraucht, den zu verstehen. Ich würde hier etwas deutlicher mit `INNER JOIN ... ON ...` arbeiten. Das würde den Select auch noch beschleunigen.

Reply:
wie besprochen

Reply:
```suggestion
        "SELECT w.uuid AS w_uuid, w.label AS label FROM webpages w INNER JOIN webpage_webpages ww ON ww.child_webpage_uuid=w.uuid WHERE ww.parent_webpage_uuid = '"
            + parentWebpageUuid + "'";
```
Sowas schwebt mir da vor :thinking:

Reply:
getestet und für gut befunden ;-)